### PR TITLE
Update for OFMCC-251 - View Notifications

### PR DESCRIPTION
**Pull request by jgstorey**

### CODE REVIEW

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 1]
```diff
@@ -1 +1,236 @@
...
+                    this.updateMessage(this.notifications[index].messageId, isRead)
+                }
+            })
+            this.initCheckBoxState()
+        },
+        /**
+         * Updates a notification to unread.
+         */
+        updateNotificationUnread(item) {
+            if (this.isRead(item)) {
```
Comment (weskubo-cgi) [inline frontend/src/components/notifications/NotificationsTab.vue:162]: I don't think this check is necessary. The button only renders when read.

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 2]
```diff
@@ -1 +1,236 @@
...
+            } else {
+                this.checkBoxClickedState = false
+            }
+        },
+        /**
+         * Updates all checked messages to read or unread.
+         */
+        updateCheckedReadUnread(isRead) {
+            this.checkBoxListState.forEach((item, index) => {
+                if (item) {
```
Comment (weskubo-cgi) [inline frontend/src/components/notifications/NotificationsTab.vue:152]: As I mentioned here you could just manipulate the item directly.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 3]
```diff
@@ -1 +1,233 @@
...
+                </template>
+                <template v-slot:bottom></template>
+            </v-data-table>
+        </v-col>
+        <v-col cols="6">
+            <v-card v-if="message" variant="flat">
+                <v-card-title class="card-title d-flex align-start flex-wrap">
+                    <div class="d-flex align-center justify-space-between w-100">
+                        <div class="d-flex align-center">
+                            <span><b>From:</b></span>&nbsp;Operating Funding Model Program
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 4]
```diff
@@ -1 +1,233 @@
...
+         * Initializes the toggle all checkbox and checkbox list state.
+         */
+        initCheckBoxState() {
+            this.checkBoxListState = Array(this.notifications.length).fill(false)
+            this.checkBoxToggleAllState = false
+        },
+        /**
+         * Indicates if the item is read or not.
+         */
+        isRead(item) {
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 5]
```diff
@@ -1 +1,233 @@
...
+                    </tr>
+                </template>
+                <template #item="{ item, index }">
+                    <tr @click="rowClickHandler(item, index)"
+                        :class="{ 'unread-message': item.selectable.isRead === false, 'highlighted': index === rowClickedIndex }">
+                        <td :class="{ 'highlighted': index === rowClickedIndex }">
+                            <v-checkbox @click="checkBoxClickHandler" v-model="checkBoxListState[index]" hide-details
+                                density="compact"></v-checkbox>
+                        </td>
+                        <td :class="{ 'highlighted': index === rowClickedIndex }">{{ item.selectable.isRead }}</td>
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 6]
```diff
@@ -1 +1,233 @@
...
+                density="compact">
+                <template v-slot:top>
+                    <div class="slot-top">
+                        <div class="flex-container">
+                            <!--div class="flex-item">
+                                <v-icon class="icon">mdi-email-plus-outline</v-icon><span class="icon-text">New
+                                    Message</span>
+                            </div-->
+                            <div class="flex-item">
+                                <v-icon @click="updateCheckedReadUnread(false)" class="icon">mdi-email-outline</v-icon><span
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

File: frontend/src/stores/messages.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import { ApiRoutes } from '@/utils/constants'
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/stores/messages.js:1]: Fixed.

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 8]
```diff
@@ -1 +1,233 @@
...
+  
+<script>
+import { mapState, mapActions } from 'pinia'
+import { useAuthStore } from '@/stores/auth'
+import { useMessagesStore } from '@/stores/messages'
+
+export default {
+    data() {
+        return {
+            notifications: [],
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

File: frontend/src/components/notifications/NotificationsTab.vue [hunk 9]
```diff
@@ -1 +1,236 @@
...
+        <v-col cols="6">
+            <v-card v-if="message" variant="flat">
+                <v-card-title class="card-title d-flex align-start flex-wrap">
+                    <div class="d-flex align-center justify-space-between w-100">
+                        <div class="d-flex align-center">
+                            <span><b>From:</b></span>&nbsp;Operating Funding Model Program
+                        </div>
+                        <div v-if="isRead(message)" class="d-flex align-center">
+                            <span @click="updateMessageUnread(message)" class="icon-text">Mark Unread</span>
+                            <v-icon @click="updateMessageUnread(message)" class="icon ml-3">mdi-email-outline</v-icon>
```
[the remark this answers is not in the document]
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

File: frontend/src/components/notifications/NotificationsTab.vue (hunk 1 above)
Comment (jgstorey) [reply frontend/src/components/notifications/NotificationsTab.vue]: Fixed.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**